packages/ics721/src/execute.rs

```diff
@@ -34,17 +41,29 @@ where
         msg: InstantiateMsg,
     ) -> StdResult<Response<T>> {
         CW721_CODE_ID.save(deps.storage, &msg.cw721_base_code_id)?;
-        PROXY.save(deps.storage, &None)?;
+        // proxy contracts are optional
+        INCOMING_PROXY.save(deps.storage, &None)?;
+        OUTGOING_PROXY.save(deps.storage, &None)?;
         PO.set_pauser(deps.storage, deps.api, msg.pauser.as_deref())?;
 
-        let proxy_instantiate = msg
-            .proxy
-            .map(|m| m.into_wasm_msg(env.contract.address))
-            .map(|wasm| SubMsg::reply_on_success(wasm, INSTANTIATE_PROXY_REPLY_ID))
-            .map_or(vec![], |s| vec![s]);
+        let mut proxies_instantiate: Vec<SubMsg<T>> = Vec::new();
+        if let Some(cii) = msg.incoming_proxy {
+            proxies_instantiate.push(SubMsg::reply_on_success(
+                cii.into_wasm_msg(env.clone().contract.address),
+                // on reply proxy contract is set in INCOMING_PROXY
+                INSTANTIATE_INCOMING_PROXY_REPLY_ID,
+            ));
+        }
+        if let Some(cii) = msg.outgoing_proxy {
+            proxies_instantiate.push(SubMsg::reply_on_success(
+                cii.into_wasm_msg(env.contract.address),
+                // on reply proxy contract is set in OUTGOING_PROXY
+                INSTANTIATE_OUTGOING_PROXY_REPLY_ID,
+            ));
+        }
```

Comment:
instantiate2 is needed for getting cw721, since cw721 is required for sending it to other chain.
here - incoming and outgoing storage - is not optional. there's no real benefit of using instantiate2 for incoming and outgoing proxy.

Reply:
Created an issue for this: https://github.com/public-awesome/cw-ics721/issues/77